webpack.common.js

```diff
@@ -1,15 +1,31 @@
 const path = require('path')
+const fs = require('fs')
+
 const HTMLWebpackPlugin = require('html-webpack-plugin')
 const FaviconsWebpackPlugin = require('favicons-webpack-plugin')
 
+const localConfig = path.resolve(__dirname, 'config-local.js')
+const defaultConfig = path.resolve(__dirname, 'config.js')
+let config
+if (fs.existsSync(localConfig)) {
+    config = localConfig
+} else if (fs.existsSync(defaultConfig)) {
+    config = defaultConfig
+} else {
+    throw `The config file is missing: ${defaultConfig}`
+}
+
 module.exports = {
     entry: path.resolve(__dirname, 'src', 'index.tsx'),
     output: {
         path: path.resolve(__dirname, 'dist'),
         filename: 'bundle.js',
     },
     resolve: {
-        alias: { '@': path.resolve(__dirname, 'src') },
+        alias: {
+            '@': path.resolve(__dirname, 'src'),
+            config$: config
```

Comment:
Here we define the webpack alias, see the docs here: https://webpack.js.org/configuration/resolve/#resolvealias

Reply:
One advantage of this approach is that the live dev server recognizes changes in the config file, i.e. we can change the api key while the app is running. We have to restart when we want to switch from `config.js` to `config-local.js`, though, because webpack only executes this code at start up.